### PR TITLE
`rustc_builtin_macros` cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
Some improvements I found while looking over this code.

r? @fee1-dead

### CODE REVIEW

File: src/tools/rustfmt/src/parse/macros/mod.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ fn check_keyword<'a, 'b: 'a>(parser: &'a mut Parser<'b>) -> Option<MacroArg> {
     None
 }
 
-pub(crate) fn parse_macro_args(
+pub fn parse_macro_args(
```
Comment (ytmimi) [inline src/tools/rustfmt/src/parse/macros/mod.rs:101]: Is the `pub(crate)` visibility within rustfmt trigger any lints? If not, could we keep this as `pub(crate)` for now?

Comment (nnethercote) [reply src/tools/rustfmt/src/parse/macros/mod.rs]: Good catch, that change is unnecessary. I have reverted it.

Comment (ytmimi) [reply src/tools/rustfmt/src/parse/macros/mod.rs]: Thank you!

**Review by fee1-dead — APPROVED** [review]
Nice! LGTM. Thanks.

**Pull request merged**